data/plugins/skill/mining/mining.rb

```diff
@@ -39,11 +51,26 @@ def start_mine(pickaxe)
   end
 
   def executeAction
-    skills = mob.skill_set
-    level = skills.get_skill(Skill::MINING).current_level
+    level = mob.skill_set.get_skill(Skill::MINING).current_level
     pickaxe = find_pickaxe
+
+    ores = @ore.id.reject { |ore| ore.level > level}
+    ore_id = if objId == 2491 then level >= 30 ? ores.last : ores.first else @ore.id.sample end
+    ore_def = ItemDefinition.lookup(ore_id.id)
+
+    name = ore_def.name.sub(/ ore$/, '').sub(/ \(.*kg\)/, '').downcase
+    granite_or_sandstone = ['sandstone', 'granite'].include? name;
+    essence = ['pure essence', 'rune essence'].include? name;
+
     mob.turn_to(@position)
 
+    # check to see if player has enough room to mine
+    if mob.inventory.free_slots.zero?
+      mob.send_message(essence ? 'Your inventory is too full to hold any more rune stones.' : "Your inventory is too full to hold any more #{name}.")
```

Comment:
mob.send_message("Your inventory is too full to hold any more #{essence ? "rune stones" : name}.")